eth/backend.go

```diff
@@ -83,18 +81,38 @@ func (b *backend) SendTransaction(ctx context.Context, tx *types.Transaction) er
 	b.nonceManager.Update(sender, tx.Nonce())
 	b.nonceManager.Unlock(sender)
 
+	var method *abi.Method
+	var methodName string
+	var txParamsString string
+
 	data := tx.Data()
-	method, ok := b.methods[string(data[:4])]
+	methodSig := data[:4]
+	abi, ok := b.abiMap[string(methodSig)]
 	if !ok {
-		method = "unknown"
+		methodName = "unknown"
+	} else {
+		m, err := abi.MethodById(methodSig)
+		if err == nil {
+			method = m
+			methodName = m.Name
+		}
+		txParams := make(map[string]interface{})
+		if err := decodeTxParams(b.abiMap[string(methodSig)], txParams, data); err != nil {
+			return err
+		}
+
+		for _, arg := range method.Inputs {
```

Comment:
Ah so it *looks* like `method` could be nil here, but if `method` is nil then `decodeTxParams()` would have already returned an error since `decodeTxParams()` also calls `abi.MethodById()` internally. 

Extracting this logic into a function could make the control flow here easier to follow and would also make this logic easier to test (would be good to add tests for the new function). Something like:

```golang
type txLog struct {
	method string
	params string
}

func (b *backend) newTxLog(tx *types.Transaction) (txLog, error) {
	data := tx.Data()
	methodSig := data[:4]
	abi, ok := b.abiMap[string(methodSig)]
	if !ok {
		return txLog{}, errors.New("unknown ABI")
	}

	m, err := abi.MethodById(methodSig)
	if err != nil {
		return txLog{}, err
	}

	methodName := m.Name
	txParams := make(map[string]interface{})
	if err := decodeTxParams(b.abiMap[string(methodSig)], txParams, data); err != nil {
		return txLog{}, err
	}

	var txParamsString string
	for _, arg := range m.Inputs {
		txParamsString += fmt.Sprintf("%v: %v  ", arg.Name, txParams[arg.Name])
	}

	txParamsString = strings.TrimSpace(txParamsString)

	return txLog{
		method: methodName,
		params: txParamsString,
	}, nil
}
```

Then, we could simplify the code in `SendTransaction()` to something like:

```golang
txLog, err := b.newTxLog(tx)
if err != nil {
    txLog.method = "unknown"
}

// Existing logging logic below
if sendErr != nil {
    ...
}

...
```

Reply:
fixed in ac6014f5